app/src/components/ArtworkList.tsx

```diff
@@ -3,37 +3,44 @@ import ArtworkItem from './ArtworkItem';
 import CardColumns from 'react-bootstrap/CardColumns';
 import { ContractProps } from '../helper/eth';
 import { useWeb3Context } from '../providers/Web3Provider';
+import { useRegistryContext } from '../providers/RegistryProvider';
 
-const ArtworkList: React.FC<ContractProps> = ({ contracts, accounts }) => {
+const ArtworkList: React.FC<ContractProps> = ({ contracts }) => {
   const [tokenIds, setTokenIds] = React.useState<Array<number>>([]);
-  const web3 = useWeb3Context();
+  const { accounts } = useWeb3Context();
+  const registry = useRegistryContext();
 
   React.useEffect(() => {
-    const artifactRegistry = web3.contracts.ArtifactRegistry;
-    const currentAccount = web3.accounts[0];
-    const updateTokenIds = (currentAccount: string): void => {
-      artifactRegistry.methods.getTokenIdsOfOwner(currentAccount)
-        .call()
-        .then((tokenIds: Array<number>) => {
-          setTokenIds(tokenIds);
-        })
-        .catch(console.log);
+    const subscribeToTransfer: VoidFunction = () => {
```

Comment:
Bit confused at the function scoping here? even if done for a good reason might be worth a refactor for clarity

Reply:
Refactor genuinely might take longer than the refactor to do event based provenance did just to pass meaningful callbacks around everywhere and not break everything